benchmark-utils_test.go

```diff
@@ -224,21 +201,62 @@ func runGetObjectBenchmark(b *testing.B, obj ObjectLayer, objSize int) {
 
 }
 
-// randomly picks a character and returns its equivalent byte array.
-func getRandomByte() []byte {
-	const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
-	// seeding the random number generator.
-	rand.Seed(time.Now().UnixNano())
-	var b byte
-	// pick a character randomly.
-	b = letterBytes[rand.Intn(len(letterBytes))]
-	return []byte{b}
+const alphabets = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+// Returns random character as byte from letterBytes.
+func getRandByte() byte {
+	return alphabets[rand.Intn(len(alphabets))]
 }
 
-// picks a random byte and repeats it to size bytes.
-func generateBytesData(size int) []byte {
-	// repeat the random character chosen size
-	return bytes.Repeat(getRandomByte(), size)
+// Returns os.File and MD5 hash string of required size of random bytes.
+func mustGetRandomReader(size int) (file *os.File, hash string) {
+	rand.Seed(time.Now().UnixNano())
+	hasher := md5.New()
+
+	// Create temporary file.
+	tmpfile, err := ioutil.TempFile("", "minio-bench")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(tmpfile.Name())
+
+	var buf []byte
+	if size < readSizeV1 {
+		buf = bytes.Repeat([]byte{getRandByte()}, size)
+	} else {
+		buf = bytes.Repeat([]byte{getRandByte()}, readSizeV1)
+	}
+
+	// Write to the temporary file.
+	bytesLeft := size
+	n := 0
+	for bytesLeft > 0 {
+		if bytesLeft > readSizeV1 {
+			n = readSizeV1
+		} else {
+			n = bytesLeft
+		}
+
+		if _, err = tmpfile.Write(buf[:n]); err != nil {
```

Comment:
Can you benchmark and compare the performance with `bufio.NewWriter`, its known to be more performant.